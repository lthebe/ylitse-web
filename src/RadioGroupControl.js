import React from 'react';
import PropTypes from 'prop-types';
import { FormControl, FormLabel, FormControlLabel } from 'material-ui/Form';
import Radio, { RadioGroup } from 'material-ui/Radio';

const RadioGroupControl = ({
    name, label, value, options, disabled, required, className, onChange,
}) => (
    <FormControl
        component="fieldset"
        disabled={disabled}
        required={required}
        className={className}
    >
        <FormLabel component="legend">{label}</FormLabel>
        <RadioGroup
            name={name}
            value={value}
            row
            onChange={onChange}
        >
            {options.map(option => (
                <FormControlLabel
                    key={option}
                    label={option}
                    value={option}
                    control={<Radio />}
                />
            ))}
        </RadioGroup>
    </FormControl>
);

RadioGroupControl.propTypes = {
    name: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
    options: PropTypes.arrayOf(PropTypes.string).isRequired,
    disabled: PropTypes.bool,
    required: PropTypes.bool,
    className: PropTypes.string,
    onChange: PropTypes.func.isRequired,
};

RadioGroupControl.defaultProps = {
    disabled: false,
    required: false,
    className: '',
};

export default RadioGroupControl;
