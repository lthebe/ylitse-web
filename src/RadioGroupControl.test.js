import React from 'react';
import { shallow } from 'enzyme';
import RadioGroupControl from './RadioGroupControl';

describe('RadioGroupControl', () => {
    const onChange = jest.fn();
    const props = {
        name: 'test',
        label: 'Test label',
        value: 'One',
        options: ['One', 'Two'],
        onChange,
    };

    test('renders correctly', () => {
        const wrapper = shallow(<RadioGroupControl {...props} />);

        expect(wrapper).toMatchSnapshot();
    });
});
