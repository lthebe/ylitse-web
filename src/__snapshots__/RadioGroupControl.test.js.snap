// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroupControl renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RadioGroupControl
    className=""
    disabled={false}
    label="Test label"
    name="test"
    onChange={[MockFunction]}
    options={
      Array [
        "One",
        "Two",
      ]
    }
    required={false}
    value="One"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(FormLabel)
          component="legend"
        >
          Test label
        </WithStyles(FormLabel)>,
        <RadioGroup
          name="test"
          onChange={[MockFunction]}
          row={true}
          value="One"
        >
          <WithStyles(FormControlLabel)
            control={<WithStyles(Radio) />}
            label="One"
            value="One"
          />
          <WithStyles(FormControlLabel)
            control={<WithStyles(Radio) />}
            label="Two"
            value="Two"
          />
        </RadioGroup>,
      ],
      "className": "",
      "component": "fieldset",
      "disabled": false,
      "required": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Test label",
          "component": "legend",
        },
        "ref": null,
        "rendered": "Test label",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              label="One"
              value="One"
            />,
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              label="Two"
              value="Two"
            />,
          ],
          "name": "test",
          "onChange": [MockFunction],
          "row": true,
          "value": "One",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "One",
            "nodeType": "class",
            "props": Object {
              "control": <WithStyles(Radio) />,
              "label": "One",
              "value": "One",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Two",
            "nodeType": "class",
            "props": Object {
              "control": <WithStyles(Radio) />,
              "label": "Two",
              "value": "Two",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(FormLabel)
            component="legend"
          >
            Test label
          </WithStyles(FormLabel)>,
          <RadioGroup
            name="test"
            onChange={[MockFunction]}
            row={true}
            value="One"
          >
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              label="One"
              value="One"
            />
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              label="Two"
              value="Two"
            />
          </RadioGroup>,
        ],
        "className": "",
        "component": "fieldset",
        "disabled": false,
        "required": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Test label",
            "component": "legend",
          },
          "ref": null,
          "rendered": "Test label",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(FormControlLabel)
                control={<WithStyles(Radio) />}
                label="One"
                value="One"
              />,
              <WithStyles(FormControlLabel)
                control={<WithStyles(Radio) />}
                label="Two"
                value="Two"
              />,
            ],
            "name": "test",
            "onChange": [MockFunction],
            "row": true,
            "value": "One",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "One",
              "nodeType": "class",
              "props": Object {
                "control": <WithStyles(Radio) />,
                "label": "One",
                "value": "One",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Two",
              "nodeType": "class",
              "props": Object {
                "control": <WithStyles(Radio) />,
                "label": "Two",
                "value": "Two",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
